### PR TITLE
Make a mutation function configurable

### PR DESCRIPTION
**Pull request by michall-m**
Adds a mutation function to configurable functions; implements default mutation function.

### CODE REVIEW

File: src/conf_functions.rs [hunk 1]
```diff
@@ -10,4 +11,5 @@ pub trait ConfigurableFunctions<const N: usize> {
     fn fitness(args: &[f64; N]) -> f64;
     fn combat(this_agent_fitness: f64, other_agent_fitness: f64) -> f64;
     fn reproduction_chance(energy: u32) -> ReproductionChance;
+    fn mutatate(genes: &mut [f64; N]);
```
Comment (michall-m) [inline src/conf_functions.rs]: mutate*

File: src/conf_functions/mutation_functions.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+use rand::{thread_rng, Rng};
+
+pub fn default_mutation<const N: usize>(genes: &mut [f64; N], domain: [(f64, f64); N]) {
```
Comment (jerzywilczek) [inline src/conf_functions/mutation_functions.rs]:
```suggestion
pub fn default_mutation<const N: usize>(genes: &mut [f64; N], domain: &[(f64, f64); N]) {
```

File: examples/rosenbrock_example.rs [hunk 3]
```diff
@@ -33,6 +35,10 @@ impl ConfigurableFunctions<2> for Configuration {
     fn reproduction_chance(energy: u32) -> emas_rs::ReproductionChance {
         default_reproduction_chance(energy)
     }
+
+    fn mutatate(genes: &mut [f64; 2]) {
```
Comment (jerzywilczek) [inline examples/rosenbrock_example.rs]: change `mutatate` to `mutate` everywhere please

**Review by jerzywilczek — CHANGES_REQUESTED** [review]
Would be cool to find a nicer way, but I don't see one right now.

**Review by stlenyk — APPROVED** [review]

**Pull request merged**